en/docs/install-and-setup/upgrading-from-wso2-ei/upgrading-from-ei-6.1.0.md

```diff
@@ -1012,10 +1012,72 @@ Follow the instructions given below if you have used a **custom log4j component*
+To migrate the encrypted passwords from EI 6.1.0, you need to first re-encrypt (using OAEP) all the encrypted information 
+in the database. Then, you can obtain the plain-text passwords by following the normal procedure of encrypting secrets 
+in the Micro Integrator. See [Encrypting Secrets]({{base_path}}/install-and-setup/setup/mi-setup/security/encrypting_plain_text) 
+for instructions.
 
-In case you need to obtain the plain-text passwords by decrypting the encrypted passwords in the EI 6.1.0,
-you can use the [password decryption tool](https://github.com/wso2-docs/WSO2_EI/tree/master/migration-client).
+!!! Info API-M 4.0.0 uses OAEP for data encryption in addition to the RSA algorithm 
+(which is used in the ESB profile of EI 6.1.0). Therefore, the internally-encrypted data in your current databases (such as 
+datasource configurations, syslog passwords, user store configurations, keystore registry entries, service security policies, 
+event publisher configurations, event receiver configurations, and event sink configurations), as well as data encrypted using 
+secure vault (such as plain text passwords in configuration files and synapse configurations) should be re-encrypted using OAEP.
+
+#### Step 1: Re-encrypt all internally-encrypted data using OAEP
+
+First, let's re-encrypt all the internally-encrypted data by using OAEP.
+
+1. Get the latest update for your existing EI 6.1.0 by using [WSO2 Update Manager](https://updates.docs.wso2.com/en/latest/updates/overview/). This will give you a new EI distribution with the latest updates.
+
+	!!! Info
+		Note that you need a valid [WSO2 subscription](https://wso2.com/subscription) to use updates in a production environment.
+		
+2. Connect the WUM-updated EI distribution to your existing databases (which are used for registry data and user management data):
+
+	??? note "Connecting to the database"
+	
+		a. Open the `master-datasources.xml` file (stored in the `<WUM_UPDATED_EI_6.1.0_HOME>/conf/datasources/` directory) and update the parameters given below.
+		
+		   By default, registry and user management data are stored in one database and is configured in the `master-datasources.xml` file. If you have separate databases for registry and user management data, you may have separate datasource configurations.
+		   
+		   | Element          | Description   |
+		   | ---------------- | ------------- |
+		   | driverClassName  | The class name of the database driver. |
+		   | url  | The URL of the database. |
+		   | username and password  | The name and password of the database user. |
+
+		   
+		b. Open the `registry.xml` file (stored in the `<WUM_UPDATED_EI_6.1.0_HOME>/conf` directory) and specify the datasource name.
+
+		   ```xml
+		   <dbConfig name="wso2registry">   
+               <dataSource>jdbc/MY_DATASOURCE_NAME</dataSource>
+           </dbConfig>
+           ```
```

Comment:
Please check if the formatting works in the docs when you generate.

Reply:
checked